packages/core/strapi/src/utils/fetch.ts

```diff
@@ -1,28 +1,31 @@
-import nodeFetch, { RequestInit, Response } from 'node-fetch';
 import { HttpsProxyAgent, HttpsProxyAgentOptions } from 'https-proxy-agent';
+import stringify from 'fast-json-stable-stringify';
 
 import type { Strapi } from '@strapi/types';
 
 export interface Fetch {
   (url: string, options: RequestInit): Promise<Response>;
-  agent?: HttpsProxyAgent;
+  dispatcher?: HttpsProxyAgent;
 }
 
+// Create a wrapper for Node's Fetch API that applies a global proxy
 export function createStrapiFetch(strapi: Strapi): Fetch {
-  function fetch(url: string, options: RequestInit) {
-    return nodeFetch(url, {
-      ...(fetch.agent ? { agent: fetch.agent } : {}),
+  function strapiFetch(url: string, options: RequestInit) {
+    const fetchOptions = {
+      ...(strapiFetch.dispatcher ? { dispatcher: strapiFetch.dispatcher } : {}),
       ...options,
-    });
+    };
+    strapi.log.debug(`Fetch request for ${url} with ${stringify(fetchOptions)}`);
```

Comment:
After playing with this for a while, it feels like too much logging, but I think that's because yarn develop should not include debug messages. The server making a remote request seems like something worth logging at the debug level.